packages/grafana-ui/src/components/Monaco/CodeEditor.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { withTheme } from '../../themes';
+import { Themeable } from '../../types';
+import MonacoEditor from 'react-monaco-editor';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
+
+export enum CodeEditorUpdateAction {
+  Blur = 'blur',
+  Save = 'save',
+}
+
+export interface Props extends Themeable {
+  text: string;
+
+  language: string;
+
+  readOnly?: boolean;
+
+  onChange: (text: string, action: CodeEditorUpdateAction) => void;
```

Comment:
this is only called on blur or `crtl+s`... so maybe it should be called "onUpdate" or somethign else?